### PR TITLE
Enable fetching of (and running with) FV3GFS external model data in netcdf format from NOAA HPSS

### PR DESCRIPTION
**Pull request by gsketefian**
## DESCRIPTION OF CHANGES: 
This PR adds the capability to fetch FV3GFS external model data in netcdf format from NOAA HPSS.  Main changes:
1) Modify workflow scripts where necessary to include stanzas for FV3GFS_FILE_FMT_ICS and FV3GFS_FMT_LBCS being set to "netcdf" (currently, only "nemsio" and "grib2" are valid values for these parameters).
2) Add a WE2E test for this capability (config.get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2021062000.sh).

## TESTS CONDUCTED:
Successfully ran the following 3 WE2E tests (including the new one introduced here) on Hera:
1) get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_grib2_2021010100
2) get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021010100
3) get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2021062000

## ISSUE (optional):
This resolves Issue #470.

### CODE REVIEW

File: ush/get_extrn_mdl_file_dir_info.sh [hunk 1]
```diff
@@ -889,33 +855,20 @@ has not been specified for this external model:
...
+        last_fhr_in_netcdfa="39"
+        first_lbc_fhr="${lbc_spec_fhrs[0]}"
+        last_lbc_fhr="${lbc_spec_fhrs[-1]}"
         if [ "${last_lbc_fhr}" -le "${last_fhr_in_netcdfa}" ]; then
-          arcv_fns=""
+          arcv_fns="${arcv_fns}gfs_nca"
         elif [ "${first_lbc_fhr}" -gt "${last_fhr_in_netcdfa}" ]; then
-          arcv_fns=""
+          arcv_fns="${arcv_fns}gfs_ncb"
         else
```
Comment (JeffBeck-NOAA) [inline ush/get_extrn_mdl_file_dir_info.sh:867]: So the "a" and "b" files aren't like grib2 pgrb2 and pgrb2b files.  These are like the nemsio "a" and "b" files where they're split based on forecast length?

File: scripts/exregional_make_lbcs.sh [hunk 2]
```diff
@@ -324,13 +324,18 @@ case "${EXTRN_MDL_NAME_LBCS}" in
...
   elif [ "${FV3GFS_FILE_FMT_LBCS}" = "grib2" ]; then
     external_model="GFS"
     fn_grib2="${EXTRN_MDL_FNS[0]}"
     input_type="grib2"
+  elif [ "${FV3GFS_FILE_FMT_LBCS}" = "netcdf" ]; then
+    external_model="FV3GFS"
+    input_type="gaussian_netcdf"     # For FV3GFS data on a Gaussian grid in netcdf format.
+    tracers_input="[\"spfh\",\"clwmr\",\"o3mr\",\"icmr\",\"rwmr\",\"snmr\",\"grle\"]"
+    tracers="[\"sphum\",\"liq_wat\",\"o3mr\",\"ice_wat\",\"rainwat\",\"snowwat\",\"graupel\"]"
   fi
```
Comment (JeffBeck-NOAA) [inline scripts/exregional_make_lbcs.sh:339]: I guess this section was missed in the template that was created (looks like it was only in the IC file).

File: scripts/exregional_make_ics.sh [hunk 3]
```diff
@@ -408,31 +407,25 @@ case "${EXTRN_MDL_NAME_ICS}" in
 "FV3GFS")
   if [ "${FV3GFS_FILE_FMT_ICS}" = "nemsio" ]; then
     external_model="FV3GFS"
+    input_type="gaussian_nemsio"     # For FV3GFS data on a Gaussian grid in nemsio format.
     tracers_input="[\"spfh\",\"clwmr\",\"o3mr\",\"icmr\",\"rwmr\",\"snmr\",\"grle\"]"
     tracers="[\"sphum\",\"liq_wat\",\"o3mr\",\"ice_wat\",\"rainwat\",\"snowwat\",\"graupel\"]"
```
Comment (JeffBeck-NOAA) [inline scripts/exregional_make_ics.sh:412]: Let's hope that the tracers haven't changed from nemsio to netcdf files!

File: scripts/exregional_make_ics.sh [hunk 4]
```diff
@@ -408,31 +407,25 @@ case "${EXTRN_MDL_NAME_ICS}" in
...
     input_type="grib2"
     convert_nst=False
-  
   elif [ "${FV3GFS_FILE_FMT_ICS}" = "netcdf" ]; then
-
     external_model="FV3GFS"
-
+    input_type="gaussian_netcdf"     # For FV3GFS data on a Gaussian grid in netcdf format.
     tracers_input="[\"spfh\",\"clwmr\",\"o3mr\",\"icmr\",\"rwmr\",\"snmr\",\"grle\"]"
     tracers="[\"sphum\",\"liq_wat\",\"o3mr\",\"ice_wat\",\"rainwat\",\"snowwat\",\"graupel\"]"
```
Comment (JeffBeck-NOAA) [inline scripts/exregional_make_ics.sh:425]: Let's hope that the tracers haven't changed from nemsio to netcdf files!

File: ush/get_extrn_mdl_file_dir_info.sh (hunk 1 above)
Comment (gsketefian) [reply ush/get_extrn_mdl_file_dir_info.sh:867]: Yes, they're exactly analogous to the nemsio files.

File: scripts/exregional_make_lbcs.sh (hunk 2 above)
Comment (gsketefian) [reply scripts/exregional_make_lbcs.sh:339]: Yep.  There was one other place I had to add the "netcdf" case.  But it works now.

File: scripts/exregional_make_ics.sh (hunk 4 above)
Comment (gsketefian) [reply scripts/exregional_make_ics.sh:425]: Well it ran fine, so I assume it's ok.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Pull request merged**